testing-common/src/main/groovy/io/opentelemetry/instrumentation/test/TestOpenTelemetryInstaller.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.test;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.trace.TracerProvider;
+import io.opentelemetry.api.trace.propagation.HttpTraceContext;
+import io.opentelemetry.context.propagation.ContextPropagators;
+import io.opentelemetry.context.propagation.DefaultContextPropagators;
+import io.opentelemetry.javaagent.tooling.OpenTelemetryInstaller;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.trace.SpanProcessor;
+import java.lang.reflect.Method;
+
+public class TestOpenTelemetryInstaller extends OpenTelemetryInstaller {
+
+  private final SpanProcessor spanProcessor;
+
+  public TestOpenTelemetryInstaller(SpanProcessor spanProcessor) {
+    this.spanProcessor = spanProcessor;
+  }
+
+  @Override
+  public void afterByteBuddyAgent() {
+    // TODO this is probably temporary until default propagators are supplied by SDK
+    //  https://github.com/open-telemetry/opentelemetry-java/issues/1742
+    //  currently checking against no-op implementation so that it won't override aws-lambda
+    //  propagator configuration
+    if (OpenTelemetry.getGlobalPropagators()
+        .getTextMapPropagator()
+        .getClass()
+        .getSimpleName()
+        .equals("NoopTextMapPropagator")) {
+      // Workaround https://github.com/open-telemetry/opentelemetry-java/pull/2096
+      setGlobalPropagators(
+          DefaultContextPropagators.builder()
+              .addTextMapPropagator(HttpTraceContext.getInstance())
+              .build());
+    }
+    OpenTelemetrySdk.getGlobalTracerManagement().addSpanProcessor(spanProcessor);
+  }
+
+  // Workaround https://github.com/open-telemetry/opentelemetry-java/pull/2096
+  public static void setGlobalPropagators(ContextPropagators propagators) {
```

Comment:
Oh we should be able to remove this now since we're on 0.12.0

Reply:
I can clean it up in a separate PR

Reply:
@pavolloffay I'll do this in #1643

Reply:
@anuraaga did this in #1924 👍